astropy/coordinates/representation.py

```diff
@@ -2659,6 +2660,13 @@ def __init__(self, d_lon, d_lat=None, copy=True):
         if not self._d_lon.unit.is_equivalent(self._d_lat.unit):
             raise u.UnitsError('d_lon and d_lat should have equivalent units.')
 
+    @classmethod
+    def from_cartesian(cls, other, base):
```

Comment:
If this is new API, probably wanna re-milestone to 4.3.

Reply:
No, it is actually a bug fix... I had not quite realized that the base class method would not work correctly for the `UnitSphericalDifferental`. Though if difficult to backport, it can be for 4.3.

Reply:
Well, if this bug is in LTS, I think the current milestone is fine and we'll leave it up to the release manager(s) to decide later. Thanks!